### ISSUE TITLE
Rule Change: (max-statements-per-line) reconcile existing conditional handling with `curly` rule

### ISSUE THREAD

**Issue by thw0rted** [eslint/eslint#15143]
### What rule do you want to change?

max-statements-per-line

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
if (a) a *= 1;
if (a) { a *= 1; }
```

### What does the rule currently do for this code?

Flags the second line, but not the first

### What will the rule do after it's changed?

Don't flag either line

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

Example from #5926.  Per [this older comment](https://github.com/eslint/eslint/issues/5926#issuecomment-215609534), "keeping [a short conditional statement] on a single line is a very common practice (guard clauses, etc.).", and as long as the statement is not set in a block, the rule won't count it.  (I didn't really understand from that comment why braces aren't acceptable for this "very common practice".)

However, this conflicts with [the `curly` rule](https://eslint.org/docs/rules/curly) which means that you must put `a *= 1` on its own line, if both rules are set (and the max-per-line value is 1) -- even though [brace-style](https://eslint.org/docs/rules/brace-style) would otherwise be happy with it on the same line, provided you use the fairly common config of `["1tbs", {allowSingleLine: true}]`.

I'd like to see some way to make these rules all play nice together.  Ideally, it would be automatic, such that `curly` and `max-statements-per-line` didn't make conflicting requirements, but I'm sure that would be a semver issue, so I'd take a flag for `max-statements-per-line` that tells it not to count single braced statements after a conditional.
